### PR TITLE
Audit History Exports

### LINKED ISSUE #2637: Track events related to work orders history exports
Log an initiated event when a user exports work order history for a project.

```
Event -> created
Actor -> Name of the user with email
Event Description -> created: Work order history export
```

### PR DESCRIPTION
**Pull request by elias-ba**
### Description

This PR implements audit events for the history export functionality. Anytime the user clicks the export button, an audit event is triggered to log the action. The audit logs track the initiation of export, providing information on the project_id, user_id, and any relevant search parameters used during the export. 

Closes #2637 

### Validation steps

1. Go to the history page
2. Initiate a history export (see cloud icon in table footer)
3. Navigate to the audit events page
4. Note the history_export audit log in the audits table

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [x] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: lib/lightning/invocation.ex [hunk 1]
```diff
@@ -650,4 +652,59 @@ defmodule Lightning.Invocation do
...
+    Ecto.Multi.new()
+    |> Ecto.Multi.run(:audit, fn _repo, _changes ->
+      ExportAudit.event(
+        "started",
+        project.id,
+        user.id,
+        %{},
+        %{search_params: search_params}
+      )
+      |> ExportAudit.save()
```
Comment (rorymckinley) [inline lib/lightning/invocation.ex:690]: @elias-ba This is interesting - in my work with the snapshot events I have (IIRC) been doing a `Multi.insert` and then returning the changeset from the `event` call.  

Would you mind sharing any advantages that you think can be found when using `Multi.run` with `ExportAudit.save`, please? (Asking for a friend :)).

File: lib/lightning_web/live/run_live/index.ex [hunk 2]
```diff
@@ -480,40 +479,30 @@ defmodule LightningWeb.RunLive.Index do
...
+        {:noreply,
+         socket
+         |> assign(:show_export_modal, false)
+         |> put_flash(
+           :info,
+           "History export started successfully. You will be notified by email after completion."
+         )}
+
+      {:error, :export_job, _reason, %{project_file: project_file}} ->
+        Projects.File.mark_failed(project_file)
```
Comment (rorymckinley) [inline lib/lightning_web/live/run_live/index.ex:498]: @elias-ba I like what you did when you extracted the login into `Invocation.export_workorders`  but now I am wondering if calling `mark_failed` should not live in there rather? What do you think?

File: test/lightning/invocation_test.exs [hunk 3]
```diff
@@ -1125,4 +1128,37 @@ defmodule Lightning.InvocationTest do
...
+      {:ok, project: project, user: user, search_params: search_params}
+    end
+
+    test "initiates a work orders export successfully and logs the audit event",
+         %{
+           project: project,
+           user: user,
+           search_params: search_params
+         } do
+      Oban.Testing.with_testing_mode(:manual, fn ->
```
Comment (rorymckinley) [inline test/lightning/invocation_test.exs:1147]: @elias-ba Is there value in asserting that the Oban job has been correctly enqueued?

File: test/lightning/invocation_test.exs [hunk 4]
```diff
@@ -10,6 +11,8 @@ defmodule Lightning.InvocationTest do
 
   require SearchParams
 
+  setup :verify_on_exit!
```
Comment (rorymckinley) [inline test/lightning/invocation_test.exs:14]: @elias-ba Can you please put me out of my misery and explain why this is here :) ?

File: lib/lightning/workorders/export_audit.ex [hunk 5]
```diff
@@ -0,0 +1,9 @@
+defmodule Lightning.WorkOrders.ExportAudit do
+  @moduledoc """
+  Module to log history export actions as audit events.
+  """
+  use Lightning.Auditing.Audit,
+    repo: Lightning.Repo,
+    item: "history_export",
+    events: ["started"]
```
Comment (rorymckinley) [inline lib/lightning/workorders/export_audit.ex:8]: @elias-ba As a matter of interest, are you planning on adding an event for completion of the export?

**Comment by rorymckinley** [comment]
@elias-ba Looks good! Also, thanks for explaining how to trigger the export - that could have been hours of my life right there!

I asked some dumb questions :) and I am happy to approve but I notice that Codecov appears to be unhappy with your changes. What is the policy in this case?

**Comment by rorymckinley** [comment]
@elias-ba I am not sure what official policy is re: the use of Mock, but how would you feel about mocking `Invocation.export_workorders` and have it spit out whichever failure response you want? With my limited context, that is where I would start.

**Comment by elias-ba** [comment]
Hey @rorymckinley I tried that. But because mocking doesn't really execute the steps of the multi in the `Invocation.export_workorders` function, CodeCov will just yell at me and tell me that these steps are not covered. I don't know if this me doing it wrong !? But I discussed with @stuartc and he thinks this is not a big deal (at least doesn't worth investing more time) as it's just complaining about the patch. What do you think ?

File: test/lightning/invocation_test.exs (hunk 3 above)
Comment (elias-ba) [reply test/lightning/invocation_test.exs:1147]: This is the best solution I found to test `export_workorders` function without having Oban to execute the job behind the scenes.

File: lib/lightning/invocation.ex (hunk 1 above)
Comment (elias-ba) [reply lib/lightning/invocation.ex:690]: I think this is just me missing the high value of using multis. I should use multi insert, thanks for the catch I will update it. My implementation is redundant

File: lib/lightning_web/live/run_live/index.ex (hunk 2 above)
Comment (elias-ba) [reply lib/lightning_web/live/run_live/index.ex:498]: I think it's a good idea. Updating the project file should be a responsibility of the presentation layer. I updated it.

File: test/lightning/invocation_test.exs (hunk 4 above)
Comment (elias-ba) [reply test/lightning/invocation_test.exs:14]: Thanks for the catch. I initially tried to mock the `export_workorders` func. Then forgot to remove that line

File: lib/lightning/workorders/export_audit.ex (hunk 5 above)
Comment (elias-ba) [reply lib/lightning/workorders/export_audit.ex:8]: I initially was tracking the enqueued, started, and completed events but then @stuartc mentioned that what we are really looking at here is being able to know who initiated a history export. Not tracking the workflow of an export. Stu any idea on this ?

Comment (rorymckinley) [reply lib/lightning/workorders/export_audit.ex:8]: @elias-ba Ah, , no worries, was just curious.

File: test/lightning/invocation_test.exs (hunk 3 above)
Comment (rorymckinley) [reply test/lightning/invocation_test.exs:1144]: @elias-ba Apologies, I worded that poorly. What I should have said is "How do you feel about adding an `assert_enqueued` to this test to validate the the Oban job has been correctly enqueued?

**Comment by rorymckinley** [comment]
@elias-ba As far as I can tell, CodeCov is happy with the tests that you have for `Invocation.export_workorders` so if you only mock `Invocation.export_workorders` for some of the tests of index.ex, I think you may get to have your cake and eat it too :). Of course, this is me chipping in from the side, so happy to pair up with you and take a stab at this if you like?

As for importance of the tests: my principle is that I will always try and add coverage if it is not too expensive to do so. However, 'expensive' is always a relative term and I do not have enough context of the broader OpenFn goings-on to make an educated call on how expensive these tests are, so I am going to take the coward's route of "I dunno". Maybe we could try a timeboxed attempt?

Thank for the feedback on my dumb questions :) - as the tests were the only thing that I was really unsure of ito approval, I  am approving.

**Review by rorymckinley — APPROVED** [review]

**Review by stuartc — APPROVED** [review]

**Comment by stuartc** [comment]
Thanks @elias-ba, gonna merge now. Small change I made was the name of the event, from `started` to `requested`; felt better to me when looking at it from the audit perspective.

**Pull request merged**